src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.Escape.cs

```diff
@@ -0,0 +1,185 @@
+        /// within an expression parsed with x mode. If future Regex features add
+        /// additional metacharacters, developers should depend on Escape to escape those
+        /// characters as well.)
+        /// </summary>
+        public static string Escape(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            return EscapeImpl(targetSpan: false, str.AsSpan(), Span<char>.Empty, out _, out _);
+        }
+
+        /// <summary>
+        /// Escapes a minimal set of metacharacters (\, *, +, ?, |, {, [, (, ), ^, $, ., #, and
+        /// whitespace) by replacing them with their \ codes. This converts a string so that
+        /// it can be used as a constant within a regular expression safely. (Note that the
+        /// reason # and whitespace must be escaped is so the string can be used safely
+        /// within an expression parsed with x mode. If future Regex features add
+        /// additional metacharacters, developers should depend on Escape to escape those
+        /// characters as well.)
+        /// </summary>
+        /// <returns>Returns the amount of chars written into the output Span.</returns>
+        public static bool TryEscape(ReadOnlySpan<char> str, Span<char> destination, out int charsWritten)
+        {
+            EscapeImpl(targetSpan: true, str, destination, out charsWritten, out bool spanSuccess);
+
+            return spanSuccess;
+        }
+
+        /// <summary>
+        /// Unescapes any escaped characters in the input string.
+        /// </summary>
+        public static string Unescape(string str)
+        {
+            if (str == null)
+                throw new ArgumentNullException(nameof(str));
+
+            return UnescapeImpl(targetSpan: false, str.AsSpan(), Span<char>.Empty, out _, out _);
+        }
+
+        /// <summary>
+        /// Unescapes any escaped characters in the input text.
+        /// </summary>
+        public static bool TryUnescape(ReadOnlySpan<char> str, Span<char> destination, out int charsWritten)
+        {
+            UnescapeImpl(targetSpan: true, str, destination, out charsWritten, out bool spanSuccess);
+
+            return spanSuccess;
+        }
+
+        /// <summary>
+        /// Escapes all metacharacters (including |,(,),[,{,|,^,$,*,+,?,\, spaces and #)
+        /// </summary>
+        private static string EscapeImpl(bool targetSpan, ReadOnlySpan<char> input, Span<char> destination, out int charsWritten, out bool spanSuccess)
+        {
+            for (int i = 0; i < input.Length; i++)
+            {
+                if (CharCategory.IsMetachar(input[i]))
+                {
+                    return EscapeImpl(targetSpan, input, i, destination, out charsWritten, out spanSuccess);
+                }
+            }
+
+            // If nothing to escape, return the input.
+            return input.CopyInput(targetSpan, destination, out charsWritten, out spanSuccess);
```

Comment:
It would be nice to return source without coping if there is nothing to escape/unescape and indicating this in an out parameter.